sentinel-adapter/sentinel-spring-webmvc-6x-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/spring/webmvc/AbstractSentinelInterceptor.java

```diff
@@ -0,0 +1,200 @@
+import com.alibaba.csp.sentinel.util.AssertUtil;
+import com.alibaba.csp.sentinel.util.StringUtil;
+
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+/**
+ * Since request may be reprocessed in flow if any forwarding or including or other action
+ * happened (see {@link jakarta.servlet.ServletRequest#getDispatcherType()}) we will only
+ * deal with the initial request. So we use <b>reference count</b> to track in 
+ * dispathing "onion" though which we could figure out whether we are in initial type "REQUEST".
+ * That means the sub-requests which we rarely meet in practice will NOT be recorded in Sentinel.
+ * <p>
+ * How to implement a forward sub-request in your action:
+ * <pre>
+ * initalRequest() {
+ *     ModelAndView mav = new ModelAndView();
+ *     mav.setViewName("another");
+ *     return mav;
+ * }
+ * </pre>
+ * 
+ * @author kaizi2009
+ * @since 1.7.1
+ */
+public abstract class AbstractSentinelInterceptor implements HandlerInterceptor {
+
+    public static final String SENTINEL_SPRING_WEB_CONTEXT_NAME = "sentinel_spring_web_context";
+    private static final String EMPTY_ORIGIN = "";
+
+    private final BaseWebMvcConfig baseWebMvcConfig;
+
+    public AbstractSentinelInterceptor(BaseWebMvcConfig config) {
+        AssertUtil.notNull(config, "BaseWebMvcConfig should not be null");
+        AssertUtil.assertNotBlank(config.getRequestAttributeName(), "requestAttributeName should not be blank");
+        this.baseWebMvcConfig = config;
+    }
+    
+    /**
+     * @param request
+     * @param rcKey
+     * @param step
+     * @return reference count after increasing (initial value as zero to be increased) 
+     */
+    private Integer increaseReferece(HttpServletRequest request, String rcKey, int step) {
+        Object obj = request.getAttribute(rcKey);
+        
+        if (obj == null) {
+            // initial
+            obj = Integer.valueOf(0);
+        }
+        
+        Integer newRc = (Integer)obj + step;
+        request.setAttribute(rcKey, newRc);
+        return newRc;
+    }
+    
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+        throws Exception {
+        try {
+            String resourceName = getResourceName(request);
+
+            if (StringUtil.isEmpty(resourceName)) {
+                return true;
+            }
+            
+            if (increaseReferece(request, this.baseWebMvcConfig.getRequestRefName(), 1) != 1) {
+                return true;
+            }
+            
+            // Parse the request origin using registered origin parser.
+            String origin = parseOrigin(request);
+            String contextName = getContextName(request);
+            ContextUtil.enter(contextName, origin);
+            Entry entry = SphU.entry(resourceName, ResourceTypeConstants.COMMON_WEB, EntryType.IN);
```

Comment:
This slot is mainly responsible for collecting resource paths and storing the call paths of these resources. The call result entry may be null. Do you need to add the following code?
```
  if (entry!= null) {
    entry. exit();
 }
```

Reply:
The code in this branch is always based on the `sentinel-spring-webmvc-adapter` module and only performs compatibility modifications.